pandas/core/apply.py

```diff
@@ -397,20 +406,66 @@ def agg_dict_like(self) -> DataFrame | Series:
         """
         return self.agg_or_apply_dict_like(op_name="agg")
 
+    def compute_dict_like(
+        self,
+        op_name: Literal["agg", "apply"],
+        selected_obj: Series | DataFrame,
+        selection: Hashable | Sequence[Hashable],
+        kwargs: dict[str, Any],
+    ) -> tuple[list[Hashable], list[Any]]:
+        obj = self.obj
+        func = cast(AggFuncTypeDict, self.func)
+        func = self.normalize_dictlike_arg(op_name, selected_obj, func)
+
+        is_non_unique_col = (
+            selected_obj.ndim == 2
+            and selected_obj.columns.nunique() < len(selected_obj.columns)
+        )
+
+        if selected_obj.ndim == 1:
+            # key only used for output
+            colg = obj._gotitem(selection, ndim=1)
+            result_data = [
+                getattr(colg, op_name)(how, **kwargs) for _, how in func.items()
+            ]
+            result_index = list(func.keys())
+        elif is_non_unique_col:
+            # key used for column selection and output
+            # GH#51099
+            result_data = []
+            result_index = []
+            for key, how in func.items():
+                indices = selected_obj.columns.get_indexer_for([key])
+                labels = selected_obj.columns.take(indices)
+                label_to_indices = defaultdict(list)
+                for index, label in zip(indices, labels):
+                    label_to_indices[label].append(index)
+
+                key_data = [
+                    getattr(selected_obj._ixs(indice, axis=1), op_name)(how, **kwargs)
+                    for label, indices in label_to_indices.items()
+                    for indice in indices
+                ]
+
+                result_index += [key] * len(key_data)
+                result_data += key_data
+        else:
+            # key used for column selection and output
+            result_data = [
+                getattr(obj._gotitem(key, ndim=1), op_name)(how, **kwargs)
+                for key, how in func.items()
+            ]
+            result_index = list(func.keys())
+
+        return result_index, result_data
```

Comment:
This just moves this section into its own method, right? (there's no change in functionality)?